core/modules/meet/client/meet-low-level.js

```diff
@@ -211,28 +254,57 @@ Template.meetLowLevel.onCreated(function () {
         }
     })
 
-    window.addEventListener(eventTypes.onUserPropertyUpdated, (e) => {
-        const { updatedProperty } = e.detail
+    window.addEventListener(eventTypes.onUserPropertyUpdated, async (e) => {
+        const { propertyName, propertyValue } = e.detail
         const localTracks = this.localTracks.get()
 
-        if ((!localTracks && localTracks.length === 0) || !updatedProperty) return
+        if ((!localTracks && localTracks.length === 0) || !propertyName) return
 
-        if (updatedProperty === 'shareAudio') {
+        if (propertyName === 'shareAudio') {
             updateTrack('audio', localTracks)
-        } else if (updatedProperty === 'shareVideo') {
+        } else if (propertyName === 'shareVideo') {
             updateTrack('video', localTracks)
+        } else if (propertyName === 'shareScreen') {
+            if (propertyValue) {
+                await meetJs.createLocalTracks({ devices: ['desktop'] }).then((tracks) => {
+                    const screenNode = document.querySelector('#video-screen-me')
+                    const screenTrack = tracks[0]
+
+                    screenTrack.attach(screenNode)
+                    this.room.addTrack(screenTrack)
+
+                    localTracks.push(screenTrack)
+                    this.localTracks.set(localTracks)
+                })
+            } else {
+                let filteredLocalTracks = localTracks.filter((track) => {
+                    if (track.getType() === 'video' && track.getVideoType() === 'desktop') {
+                        track.dispose()
+                        return false
+                    } else {
+                        return true
+                    }
+                })
+
+                this.localTracks.set(filteredLocalTracks)
```

Comment:
Hum, why? Update track does a different thing. This logic is for the shareScreen props, wich doesn't need to use updateTrack, since it's just for audio/video to mute or unmute it without removing the track.

Reply:
I just don't understand why shareScreen is handled differently and in a different place than shareAudio or shareVideo

Reply:
Because the share screen track is independant track, it will be create/delete when user want. The audio and video track are created at the initialization of the connection. When ever user toggle share audio/video, it will be muted. But when it's share screen CTA, it will be created a this moment.

Reply:
Why the different behaviors? Why don't we create the video track only when the user wants to share its video?

Reply:
Actually, I just followed the doc and never thinked about that, but yes, I think we can do that.